lib/mixins/template-stamp.html

```diff
@@ -458,25 +118,182 @@
+            template.hasAttribute('strip-whitespace');
+          this._parseTemplateNode(template.content, templateInfo, {parent: null});
+        }
+        return template._templateInfo;
+      }
+
+      // add annotations gleaned from subtree at `node` to `notes`
+      static _parseTemplateNode(node, templateInfo, nodeInfo) {
+        let noted;
+        if (node.localName == 'template' && !node.hasAttribute('preserve-content')) {
+          noted = this._parseTemplateNestedContent(node, templateInfo, nodeInfo) || noted;
+        } else if (node.localName === 'slot') {
+          // For ShadyDom optimization, indicating there is an insertion point
+          templateInfo.hasInsertionPoint = true;
+        }
+        if (node.firstChild) {
+          noted = this._parseTemplateChildNodes(node, templateInfo, nodeInfo) || noted;
+        }
+        if (node.hasAttributes && node.hasAttributes()) {
+          noted = this._parseTemplateNodeAttributes(node, templateInfo, nodeInfo) || noted;
+        }
+        return noted;
+      }
+
+      static _parseTemplateChildNodes(root, templateInfo, nodeInfo) {
+        for (let node=root.firstChild, index=0, next; node; node=next) {
+          // Wrap templates
+          if (node.localName == 'template') {
+            node = wrapTemplateExtension(node);
+          }
+          // collapse adjacent textNodes: fixes an IE issue that can cause
+          // text nodes to be inexplicably split =(
+          // note that root.normalize() should work but does not so we do this
+          // manually.
+          next = node.nextSibling;
+          if (node.nodeType === Node.TEXT_NODE) {
+            let n = next;
+            while (n && (n.nodeType === Node.TEXT_NODE)) {
+              node.textContent += n.textContent;
+              next = n.nextSibling;
+              root.removeChild(n);
+              n = next;
+            }
+            // optionally strip whitespace
+            if (templateInfo.stripWhiteSpace && !node.textContent.trim()) {
+              root.removeChild(node);
+              continue;
+            }
+          }
+          let childInfo = { index, parent: nodeInfo };
+          if (this._parseTemplateNode(node, templateInfo, childInfo)) {
+            templateInfo.nodeInfoList.push(childInfo);
+          }
+          index++;
+        }
+      }
+
+      // Recurse into nested templates
+      // 1. Parse annotations from the template and memoize them on
+      //    content._templateInfo (recurses into nested templates)
+      // 2. Remove template.content and store it in annotation list, where it
+      //    will be the responsibility of the host to set it back to the template
+      //    (this is both an optimization to avoid re-stamping nested template
+      //    children and avoids a bug in Chrome where nested template children
+      //    upgrade)
+      static _parseTemplateNestedContent(node, outerTemplateInfo, nodeInfo) {
```

Comment:
Worth trying to collapse with `_parseTemplate`? Seems like if there's an outerTemplate, you can do this content stripping.